ui-admin/src/portal/publish/diffComponents.tsx

```diff
@@ -24,77 +24,169 @@ export const VersionChangeView = ({ record }: {record: VersionedEntityChange}) =
   </div>
 }
 
+type ConfigChangesProps = {
+  configChanges: ConfigChange[],
+  selectedChanges: ConfigChange[],
+  updateSelectedChanges: (changes: ConfigChange[]) => void
+}
 /** renders a list of config changes, or "no changes" if empty */
-export const ConfigChanges = ({ configChanges }: {configChanges: ConfigChange[]}) => {
+export const ConfigChanges = ({ configChanges, selectedChanges, updateSelectedChanges }: ConfigChangesProps) => {
   if (!configChanges.length) {
     return <span className="fst-italic text-muted">no changes</span>
   }
+  /** handles add/remove from the selected changes */
+  const updateSelection = (propertyName: string, selected: boolean) => {
+    const selectionIndex = selectedChanges.findIndex(change => change.propertyName === propertyName)
+    if (selected && selectionIndex < 0) {
+      updateSelectedChanges([
+        ...selectedChanges,
+        configChanges.find(change => change.propertyName === propertyName) as ConfigChange
+      ])
+    }
+    if (!selected && selectionIndex >= 0) {
+      const newArr = [...selectedChanges]
+      newArr.splice(selectionIndex, 1)
+      updateSelectedChanges(newArr)
```

Comment:
Not requesting this change since it's more of a style preference, but if you want to do this without the intermediate variable and the in-place update, this could also be written as:

```suggestion
      updateSelectedChanges([
        ...selectedChanges.slice(0, selectionIndex),
        ...selectedChanges.slice(selectionIndex + 1)
     ])
```

Reply:
yeah, I saw that syntax on your recent panel question-order modification PR.  I think you're right that it's probably a bit cleaner.